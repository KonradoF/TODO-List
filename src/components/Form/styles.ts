import EStyleSheet from 'react-native-extended-stylesheet';

export const styles = EStyleSheet.create({
  container: {
    flex: 1,
  },
  spacing: {
    flex: 1,
    backgroundColor: 'transparent',
  },
});
