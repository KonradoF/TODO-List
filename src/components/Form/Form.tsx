import React, {ReactElement, ReactNode} from 'react';
import {KeyboardAvoidingView, Platform, ViewStyle} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import {styles} from './styles';

interface Props {
  children?: ReactNode;
  style?: ViewStyle;
}

export const Form = ({children, style}: Props): ReactElement => {
  return (
    <KeyboardAvoidingView
      contentContainerStyle={styles.container}
      style={EStyleSheet.flatten([styles.spacing, style])}
      behavior={Platform.select({ios: 'padding', android: null})}
      enabled
      keyboardVerticalOffset={Platform.select({ios: 0, android: null})}>
      {children}
    </KeyboardAvoidingView>
  );
};
