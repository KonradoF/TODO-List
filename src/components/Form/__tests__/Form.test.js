import React from 'react';
import {shallow} from 'enzyme';
import {Form} from 'src/components/Form/Form';
import {Text} from 'react-native-elements';

describe('Form', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Form />);
      expect(component).toMatchSnapshot();
    });
  });
  describe('Check render props', () => {
    let component;
    const text = 'Test';

    beforeEach(() => {
      component = shallow(
        <Form>
          <Text>{text}</Text>
        </Form>,
      );
      jest.clearAllMocks();
    });

    it('should render correct children', () => {
      const labelElement = component.find({children: text});
      expect(labelElement).toHaveLength(1);
    });
  });
});
