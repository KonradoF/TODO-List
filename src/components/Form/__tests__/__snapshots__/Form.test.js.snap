// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Rendering should match to snapshot 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
/>
`;
